src/sympc/protocol/falcon/falcon.py

```diff
@@ -63,3 +66,84 @@ def __eq__(self, other: Any) -> bool:
             return False
 
         return True
+
+    @staticmethod
+    def mul_master(x: MPCTensor, y: MPCTensor, session: Session) -> MPCTensor:
+        """Master method for multiplication.
+
+        Performs Falcon's mul implementation, gets and reshares mul results and distributes shares.
+        This needs to be improved in future, it relies on orchestrator being a trusted third party.
+        Falcon, requires parties to be able to communication between each other.
+
+        Args:
+            x (MPCTensor): Secret
+            y (MPCTensor): Another secret
+            session (Session): Session the tensors belong to
+
+        Returns:
+            shares (ReplicatedSharedTensor): Shares in terms of ReplicatedSharedTensor.
+
+        Raises:
+            ValueError: Raised when number of parties are not three.
+            NotImplementedError: Raised when implementation not present
+
+        """
+        if len(session.parties) != 3:
+            raise ValueError("Falcon requires 3 parties")
+
+        result = None
+
+        if session.protocol.security_type == "semi-honest":
+
+            z_shares = []
+            for party in range(0, len(x.share_ptrs)):
+                z_shares.append(x.share_ptrs[party] * y.share_ptrs[party])
```

Comment:
MAYBE: Also this should be in the parallel execution.

Reply:
Done